sdk/core/azure-core-amqp/vendor/azure-uamqp-c/src/connection.c

```diff
@@ -173,49 +173,64 @@ static int channel_table_allocate(CHANNEL_TABLE* channel_table, uint16_t* h)
     else
     {
         *h = 0;
-
         if (channel_table->size == channel_table->capacity)
         {
-          CHANNEL_TABLE_ENTRY* new_handles = (CHANNEL_TABLE_ENTRY*)realloc(
-              channel_table->channels, sizeof(CHANNEL_TABLE_ENTRY) * channel_table->capacity * 2);
-          if (new_handles == NULL)
-          {
-            LogError("Cannot reallocate memory for handle table");
-            result = MU_FAILURE;
-          }
-          else
-          {
-            channel_table->channels = new_handles;
-            // Ensure the newly resized handle table is empty.
-            for (uint16_t i = channel_table->size; i < channel_table->capacity * 2; i++)
+            if (channel_table->capacity == UINT16_MAX)
             {
-              channel_table->channels[i].outgoing_channel = ENDPOINT_UNUSED_CHANNEL;
-              channel_table->channels[i].incoming_channel = ENDPOINT_UNUSED_CHANNEL;
-              channel_table->channels[i].is_endpoint_live = false;
+                LogError("Cannot allocate more channels");
+                result = MU_FAILURE;
+            }
+            else
+            {
+                uint32_t new_capacity = channel_table->capacity + INITIAL_CHANNEL_TABLE_CAPACITY;
+                if (channel_table->capacity >= UINT16_MAX)
+                {
+                    new_capacity = UINT16_MAX;
+                }
+                CHANNEL_TABLE_ENTRY* new_channels = (CHANNEL_TABLE_ENTRY*)realloc(
+                    channel_table->channels, sizeof(CHANNEL_TABLE_ENTRY) * new_capacity);
+                if (new_channels == NULL)
+                {
+                    LogError("Cannot reallocate memory for handle table");
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    channel_table->channels = new_channels;
+                    channel_table->capacity = new_capacity;
+                    // Ensure the newly resized part of the handle table is empty.
+                    uint32_t i = channel_table->size;
+                    do
+                    {
+                        channel_table->channels[i].outgoing_channel = ENDPOINT_UNUSED_CHANNEL;
+                        channel_table->channels[i].incoming_channel = ENDPOINT_UNUSED_CHANNEL;
+                        channel_table->channels[i].is_endpoint_live = false;
+                        i++;
+                    } while (i < channel_table->capacity);
```

Comment:
Do you think it could be faster to memset, so if possible, this code becomes `rep movsX`?

Reply:
Why would we want to use memset if we're going to overwrite each of the fields in the struct immediately afterwards (`ENDPOINT_UNUSED_CHANNEL` is 0xffff).

Reply:
memset could initialize with non-0 value...